File: representation/matrix.cpp
#include<iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    // Initialize an adjacency matrix for the graph with O(n) space
    int adj[n+1][m+1];
    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        adj[u][v] = 1;
        adj[v][u] = 1;

        //for weighted graph, we can store the weight instead of 1
        //adj[u][v] = weight;
        //adj[v][u] = weight; //for undirected graph
        //for directed graph, we would only add one direction
        // u ---> v
        // adj[u][v] = weight; only this if the graph is directed
        // v ---> u
        // adj[v][u] = weight; only this if the graph is directed
    }
    return 0;
}

File: representation/matrix.java
package representation;
import java.io.*;

public class matrix {
    public static void main(String[] args) {
        int n = 3, m = 3;
        int adj[][] = new int[n+1][n+1];

        // edge 1---2
        adj[1][2] = 1;
        adj[2][1] = 1;

        // edge 2---3
        adj[2][3] = 1;
        adj[3][2] = 1;

        // edge 1---3
        adj[1][3] = 1;
        adj[3][1] = 1;

        adj[u][v] = 1; // edge u---v
        adj[v][u] = 1; // edge v---u

        //for weighted graph
        // adj[u][v] = weight; // edge u ---> v
        // adj[v][u] = weight; // edge v ---> u
    }
}
